### PR TITLE
Gh 18 add cognito user pool

### PR DESCRIPTION
**Pull request by m29827**

### CODE REVIEW

File: lib/authentication/user-pool.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        if (userPoolConfig.useExternalPool) {
+
+            const externalPool: IExternalPool = userPoolConfig.externalPool!;
+
+            this._userPool = cognito.UserPool.fromUserPoolId(this, KaiUserPool._userPoolId, externalPool.userPoolId);
+            this._userPoolClient = cognito.UserPoolClient.fromUserPoolClientId(this, KaiUserPool._userPoolClientId, externalPool.userPoolClientId);
+
+        } else {
+
+            this._userPool = new cognito.UserPool(this, KaiUserPool._userPoolId);
```
Comment (d47853) [inline lib/authentication/user-pool.ts]: Might be worth documenting this in the README

File: lib/authentication/user-pool.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+            this._userPoolClient = this._userPool.addClient(KaiUserPool._userPoolClientId, {
+                "userPoolClientName": KaiUserPool._userPoolClientId,
+                "generateSecret": false
+            });
+
+            const defaultPoolConfigOverrides: IDefaultPoolConfigOverrides = userPoolConfig.defaultPoolConfigOverrides!;
+
+            if (defaultPoolConfigOverrides.userPoolConfigOverrides) {
```
Comment (d47853) [inline lib/authentication/user-pool.ts]: Is it possible to add these overrides to the user pool when we create it? Might be a bit neater than changing it afterwards.

**Review by d47853 — CHANGES_REQUESTED** [review]

File: lib/authentication/user-pool.ts (hunk 1 above)
Comment (m29827) [reply lib/authentication/user-pool.ts]: Updated the documentation

File: lib/authentication/user-pool.ts (hunk 2 above)
Comment (m29827) [reply lib/authentication/user-pool.ts]: I've made this change the only caveat being that to access configuration not exposed through UserPoolProps and UserPoolClientOptions does require use of the property overrides.

**Review by d47853 — APPROVED** [review]

File: lib/authentication/user-pool.ts (hunk 2 above)
Comment (d47853) [reply lib/authentication/user-pool.ts]: Thanks. If we need to do this in future, we can.

**Pull request merged**